crates/wysiwyg/src/dom/range.rs

```diff
@@ -254,10 +262,23 @@ impl Range {
             .filter(move |l| l.node_handle.depth() == depth)
     }
 
+    pub fn locations_from_depth(
+        &self,
+        depth: usize,
+    ) -> impl Iterator<Item = &DomLocation> {
+        self.locations
+            .iter()
+            .filter(move |l| l.node_handle.depth() > depth)
```

Comment:
I think `*l.node_handle.depth()` would also work here, instead of using `move`.

Reply:
My understanding is that `move` is needed because `depth` will only live as long as the function but the closure that `filter` uses will need to run after the function returns. Will it have some bad/unintended effect other than this? 

(I have tried your suggestion but it didn't seem to work.)

Reply:
You're right, I didn't realise you were returning the iterator, not a collected version of it, so the `move` keyword is needed.